File: typescript/src/Alarm.ts
import Sensor from './Sensor';

const LOW_PRESSURE_THRESHOLD = 17;
const HIGH_PRESSURE_THRESHOLD = 21;

export default class Alarm {
    public sensor = new Sensor()

    private alarmOn = false

    check(): void {
      const psiPressureValue = this.sensor.popNextPressurePsiValue();

      if (psiPressureValue < LOW_PRESSURE_THRESHOLD || psiPressureValue > HIGH_PRESSURE_THRESHOLD) {
        this.alarmOn = true;
      }
    }

    isOn(): boolean {
      return this.alarmOn;
    }
}

File: go/alarm.go
package tirepressure

const (
	lowPressureThreshold  float64 = 17
	highPressureThreshold float64 = 21
)

type Alarm struct {
	sensor  *Sensor
	alarmOn bool
}

func (a *Alarm) Check() {
	psiPressureValue := a.sensor.PopNextPressurePsiValue()

	if psiPressureValue < lowPressureThreshold || highPressureThreshold < psiPressureValue {
		a.alarmOn = true
	}
}

func (a Alarm) IsAlarmOn() bool {
	return a.alarmOn
}

func NewAlarm() *Alarm {
	return &Alarm{sensor: NewSensor()}
}
